### PR TITLE
add HF token support through params

### PR DESCRIPTION
**Pull request by samos123**
Tested with this example: https://github.com/substratusai/substratus/pull/146

### CODE REVIEW

**Review by brandonjbjelland — APPROVED** [review]
LGTM but I bet Nick has an opinion here.

File: base/Dockerfile [hunk 1]
```diff
@@ -28,7 +28,7 @@ COPY scripts/* scripts/
 ENV TINI_VERSION v0.19.0
 ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini /tini
 RUN chmod +x /tini
-ENTRYPOINT ["/tini", "--"]
+ENTRYPOINT ["/tini", "--", "HUGGING_FACE_HUB_TOKEN=$PARAM_HUGGING_FACE_HUB_TOKEN"]
```
Comment (nstogner) [inline base/Dockerfile:31]: I would prefer to keep huggingface-things in the huggingface containers only.

**Review by nstogner — CHANGES_REQUESTED** [review]

File: base/Dockerfile (hunk 1 above)
Comment (samos123) [reply base/Dockerfile]: done, please review again

**Comment by samos123** [comment]
1 thing to note, I might file a new issue for it, our pods dont automatically pull down latest if the image was already found on the node:
```
  Normal  Pulled     8s    kubelet                                Container image "substratusai/model-loader-huggingface:pr-13" already present on machine

```

**Review by brandonjbjelland — APPROVED** [review]
🐼

File: model-loader-huggingface/Dockerfile [hunk 2]
```diff
@@ -8,4 +8,5 @@ RUN pip3 install --no-cache-dir -r requirements.txt
 COPY scripts/* scripts
 COPY src/* src/
 
+ENTRYPOINT ["/tini", "--", "/content/scripts/entrypoint.sh"]
```
Comment (nstogner) [inline model-loader-huggingface/Dockerfile:11]: Will we still get all of the benefits of tini (catching shutdown signals and passing them along?) if we call a script that calls a script that calls a executable?

Comment (samos123) [reply model-loader-huggingface/Dockerfile:11]: Hmm good question. I need to test that.

Comment (samos123) [reply model-loader-huggingface/Dockerfile:11]: I tested it by creating a python program that handles sigterm and then ran it in the image:
```
docker run -ti -v ./src:/content/src substratusai/model-loader-huggingface:pr-13 python3 /content/src/sig-handler.py
doing something in a loop ..
doing something in a loop ...
doing something in a loop ...
doing something in a loop ...
End of the program. I was killed gracefully :)
```

Then I killed the tini process
```
╭─samos@asus-arch ~/workspace/substratus-images/model-loader-huggingface ‹hf-token●› 
╰─$ ps aux | grep sig-handler
samos      95393  0.1  0.1 1327996 24928 pts/4   Sl+  19:51   0:00 docker run -ti -v ./src:/content/src substratusai/model-loader-huggingface:pr-13 python3 /content/src/sig-handler.py
root       95591  0.2  0.0   2776  1152 pts/0    Ss   19:51   0:00 /tini -- /content/scripts/entrypoint.sh python3 /content/src/sig-handler.py
root       95613  0.1  0.0  13540  8192 pts/0    S+   19:51   0:00 python3 /content/src/sig-handler.py
samos      95619  0.0  0.0   6580  2532 pts/6    S+   19:51   0:00 grep --color=auto --exclude-dir=.bzr --exclude-dir=CVS --exclude-dir=.git --exclude-dir=.hg --exclude-dir=.svn --exclude-dir=.idea --exclude-dir=.tox sig-handler
╭─samos@asus-arch ~/workspace/substratus-images/model-loader-huggingface ‹hf-token●› 
╰─$ kill 95591
kill: kill 95591 failed: operation not permitted
╭─samos@asus-arch ~/workspace/substratus-images/model-loader-huggingface ‹hf-token●› 
╰─$ sudo kill 95591                                                                
```
It all seemed to work as expected. Not sure what else to test. I'm gonna merge this.

This is the sig handler.py file I used:
```
import signal
import time

class GracefulKiller:
  kill_now = False
  def __init__(self):
    signal.signal(signal.SIGINT, self.exit_gracefully)
    signal.signal(signal.SIGTERM, self.exit_gracefully)

  def exit_gracefully(self, *args):
    self.kill_now = True

if __name__ == '__main__':
  killer = GracefulKiller()
  while not killer.kill_now:
    time.sleep(1)
    print("doing something in a loop ...")

  print("End of the program. I was killed gracefully :)")
```

**Pull request merged**